### ISSUE TITLE
Necrotic Wake, before fighting the next to last boss

### ISSUE THREAD

**Issue by bcarpio** [mikepauer/Carbonite#200]
I just ran The Necrotic Wake, before fighting the next to last boss, you're taken up into the flying Necropolis.
As soon as I landed inside I started receiving a flood of error messages, unfortunately I wasn't able to copy it at the time because I couldn't continue to do the instance until WoW asked if I wanted to Disable all my addons because of the error.

 

I tried to locate a log file but was unable to find one.  Sorry that I can't provide more details.  But I do know that it was specifically referring to Carbonite and NXMap.  It was basically indicating that a variable or index was undefined.

Copied from the https://www.curseforge.com/ page.

**Comment by Thrumbar** [comment]
This sounds like a reported error on WoWInterface forums..  Copied below..  This error does not use the latest from Git but I do not think it matters as the error line in 12271 instead of the one listed in the report and could be the zone not in database.

Everytime we reach the final two bosses above the main floor, this error constantly gets thrown

34946x Carbonite\NxMapData.lua:12270: attempt to index field '?' (a nil value)
[string "@Carbonite\NxMapData.lua"]:12270: in function `GetZoneInfo'
[string "@Carbonite\NxMap.lua"]:9855: in function `IsInstanceMap'
[string "@Carbonite\NxMap.lua"]:5328: in function `SwitchRealMap'
[string "@Carbonite\NxMap.lua"]:4512: in function `Update'
[string "@Carbonite\NxMap.lua"]:4021: in function <Carbonite\NxMap.lua:3905>

Locals:
self = <table> {
Menu_OnTrackPlyr = <function> defined @Carbonite\NxMap.lua:2960
GetInstanceID = <function> defined @Carbonite\NxMap.lua:9524
ZoneOverlays = <table> {
}
HideExtraFrames = <function> defined @Carbonite.Info\NxInfo.lua:1290
GetZoneAchievement = <function> defined @Carbonite\NxMap.lua:11144
MinimapOwnInit = <function> defined @Carbonite\NxMap.lua:1979
RestoreBlizzBountyMap = <function> defined @Carbonite\NxMap.lua:3332
MapLevels = <table> {
}
MinimapSetScale = <function> defined @Carbonite\NxMap.lua:2602
OnButToggleGuide = <function> defined @Carbonite\NxMap.lua:1669
GetWinName = <function> defined @Carbonite\NxMap.lua:1254
SetToCurrentZone = <function> defined @Carbonite\NxMap.lua:9606
GetIconStatic = <function> defined @Carbonite\NxMap.lua:8609
UpdateToolBar = <function> defined @Carbonite\NxMap.lua:1291
Menu_OnBackgndAlphaFade = <function> defined @Carbonite\NxMap.lua:2865
BlizzToggleWorldMap = <function> defined @Carbonite\NxMap.lua:3359
Menu_OnMapDebugFullCoords = <function> defined @Carbonite\NxMap.lua:2911
ClipFrameByMapType = <function> defined @Carbonite\NxMap.lua:7234
ZandalarMapBlks = <table> {
}
ResetIcons = <function> defined @Carbonite\NxMap.lua:8336
Minimap_OnEvent = <function> defined @Carbonite\NxMap.lua:2068
UnpackObjective = <function> defined @Carbonite\NxMap.lua:11392
GetWorldContinentInfo = <function> defined @Carbonite\NxMap.lua:9910
ReverseTargets = <function> defined @Carbonite\NxMap.lua:10230
GetWorldZoneScale = <function> defined @Carbonite\NxMap.lua:9977
GMenu_OnPasteLink = <function> defined @Carbonite\NxMap.lua:3089
ContCnt = 13
CreateToolBar = <function> defined @Carbonite\NxMap.lua:1261
PlyrNamesTipStr = ""
GetPlayerMapPosition = <function> defined @Carbonite\NxMap.lua:11700
RestoreWorldMap = <function> defined @Carbonite\NxMap.lua:11430
DetachWorldMap = <function> defined @Carbonite\NxMap.lua:1504
ClearTarget = <function> defined @Carbonite\NxMap.lua:10192
BGMenu_OnHelp = <function> defined @Carbonite\NxMap.lua:3125
UnregisterAllEvents = <function> defined @Ace3\CallbackHandler-1.0\CallbackHandler-1.0.lua:174
VerifySettings = <function> defined @Carbonite\NxMap.lua:313
HideMiniFrames = <function> defined @Carbonite\NxMap.lua:7121
HideExtraZoneTiles = <function> defined @Carbonite\NxMap.lua:6460
Menu_OnDetailAlpha = <function> defined @Carbonite\NxMap.lua:2861
IsInstanceMap = <function> defined @Carbonite\NxMap.lua:9851
EkMapBlks = <table> {
}
CheckWorldHotspotsType = <function> defined @Carbonite\NxMap.lua:6257
BuildPlyrLists = <function> defined @Carbonite\NxMap.lua:8748
SetMapByID = <function> defined @Carbonite\NxMap.lua:237
Menu_OnScaleRestore = <function> defined @Carbonite\NxMap.lua:2825
Menu_OnBackgndAlphaFull = <function> defined @Carbonite\NxMap.lua:2877
DraeneiXO = -3500
SetIconTypeAlpha = <function> defined @Carbonite\NxMap.lua:7977
MMMenuIFull = <table> {
}
CreateIconMenu = <function> defined @Carbonite\NxMap.lua:1301
WMDF = WMDF {
}
ScanContinents = <function> defined @Carbonite\NxMap.lua:5353
MinimapUpdateMask = <function> defined @Carbonite\NxMap.lua:2613
OLMapBlks = <table> {
}
Menu_OnPlyrFollow = <function> defined @Carbonite\NxMap.lua:2841
Menu_OnGetQuests = <function> defined @Carbonite\NxMap.lua:2951
__index = <table> {
}
I

**Comment by IrcDirk** [comment]
Shit... no time to debug it... Ill try to do the fix but u need to test it.

**Comment by bcarpio** [comment]
I can try and test it, let me know when you push the update.

**Comment by instant0** [comment]
Edit: Follow Mikepauers fix below instead to get it working properly.

If you edit "NxMapData.lua" and add the following around line 1655 (you'll see where it could go, but anywhere around this will be fine).

	[1667] = {
		Scale = 1,
		X = -341.75,
		Y = 312,
		Name = L["The Necrotic Wake - Stichwerks"],
		BaseMap = 1667,
		Alpha = .85,
		MapLevel = 1,
	},
	[1668] = {
		Scale = 1,
		X = -341.75,
		Y = 312,
		Name = L["The Necrotic Wake - Zolramus"],
		BaseMap = 1668,
		Alpha = .85,
		MapLevel = 1,
	},


This fixes the error messages, but it does not show the dungeon canvas map for these two encounters, as I couldn't figure out how.

**Comment by IrcDirk** [comment]
Im little busy right now. But ill make fix. 

@instant0 this is just a workaround, but thanks.

**Comment by mikepauer** [comment]
As it's a dungeon they shouldn't be in NxMapData, but instead in Data\zones.lua

Something like:
[1666] = L["The Necrotic Wake"] .. "|50|60|3|5|1533|40.13|55.24|5",
[1667] = L["The Necrotic Wake - Stichwerks"] .. "|50|60|3|5|1533|40.13|55.24|0",
[1668] = L["The Necrotic Wake - Zolramus"] .. "|50|60|3|5|1533|40.13|55.24|0",

**Comment by IrcDirk** [comment]
@mikepauer it can be, there is bug in auto zone adding function that i did added.

**Comment by mikepauer** [comment]
I'd try and do fixes, but right now no computer.. been waiting 2 months now for CPU delivery so I can build and get back in... hopefully not much longer so your not doing it alone.

**Comment by instant0** [comment]
@MikePauer Thanks, adding those and removing my workaround fixed it _and_ added the map information correctly.
Just tested it now.

**Comment by IrcDirk** [comment]
@instant0 yes this needs to work, i will add this to Carbonite. But i need to fix this automatic zone adding too ;)
